group.go

```diff
@@ -53,11 +53,18 @@ type Scalar interface {
 	// Set to a fresh random or pseudo-random scalar
 	Pick(rand cipher.Stream) Scalar
 
-	// SetBytes will take bytes and create a scalar out of it
+	// SetBytes sets the scalar from a big-endian byte-slice,
+	// reducing if necessary to the appropriate modulus.
 	SetBytes([]byte) Scalar
 
-	// Bytes returns the raw internal representation
+	// Bytes returns a big-Endian representation of the scalar
 	Bytes() []byte
+
+	// Allow or disallow use of faster variable-time implementations
+	// of operations on this Point, returning the old flag value.
+	// This flag always defaults to false (constant-time only)
+	// in implementations that can provide constant-time operations.
+	SetVarTime(varTime bool) bool
```

Comment:
I just realized that we are *not* providing the variable time implementation anymore for `Scalar`s in `edwards25519/` anymore.  The simplest way to do it would be to add `if s.varTime { .. } else { ... }` in every method of the `Scalar`... Having this "dual state" / dual implementation in one struct does not look very nice at all and has its pile of drawbacks (maintability, readability mainly). May I propose an alternative approach ;) ?
What about if we change the signature to 
```go
type Scalar interface {
  ...
  // VarTime returns a Scalar using variable time arithmetic operations if varTime is true. 
  // It returns a constant time implementation of a Scalar if varTime is false. 
  VarTime(varTime bool) Scalar // plus the old value if needed ?
}
```
so in our implementation we can have a clear distinction between the two implementations
```go
type scalarVarTime mod.Int
type scalarConstTime struct  {
   v [32]byte
}
```
I see at least two main advantages here:
1. API-wise: 
  - not keeping track on "which state are we on" for a specific scalar.  
  - consistency with other methods that returns the result as Scalar 
2. Implementation-wise: it's much cleaner and easier to code separate implementation, one vartime and one const. time. If I were a new developer hacking on this library, I'd find that much more elegant than having `if / else ` in every methods.